Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidStringFormatInInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp,
+		Name = nameof(AvoidStringFormatInInterpolatedStringCodeFixProvider)), Shared]
+	public class AvoidStringFormatInInterpolatedStringCodeFixProvider :
+		SingleDiagnosticCodeFixProvider<InterpolationSyntax>
+	{
+		protected override string Title => "Remove redundant string.Format call";
+
+		protected override DiagnosticId DiagnosticId =>
+			DiagnosticId.AvoidStringFormatInInterpolatedString;
+
+		protected override async Task<Document> ApplyFix(Document document, InterpolationSyntax node,
+			ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+		{
+			SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken);
+
+			// Validate and extract string.Format information
+			if (!TryExtractStringFormatInfo(node, out InvocationExpressionSyntax invocation, out var formatString, out ArgumentSyntax[] arguments))
+			{
+				return document;
+			}
+
+			// Verify this is actually a string.Format call
+			if (!await IsStringFormatInvocation(document, invocation, cancellationToken))
+			{
+				return document;
+			}
+
+			// Find the parent interpolated string and apply the transformation
+			return await ApplyInterpolatedStringTransformation(document, root, node, formatString, arguments);
+		}
+
+		private bool TryExtractStringFormatInfo(InterpolationSyntax node, out InvocationExpressionSyntax invocation, out string formatString, out ArgumentSyntax[] arguments)
+		{
+			invocation = null;
+			formatString = null;
+			arguments = null;
+
+			// Get the string.Format invocation from the interpolation
+			if (node.Expression is not InvocationExpressionSyntax inv)
+			{
+				return false;
+			}
+
+			invocation = inv;
+
+			// Extract format string and arguments
+			if (invocation.ArgumentList.Arguments.Count < 1)
+			{
+				return false;
+			}
+
+			ArgumentSyntax formatArgument = invocation.ArgumentList.Arguments[0];
+			if (formatArgument.Expression is not LiteralExpressionSyntax formatLiteral ||
+				!formatLiteral.Token.IsKind(SyntaxKind.StringLiteralToken))
+			{
+				return false;
+			}
+
+			formatString = formatLiteral.Token.ValueText;
+			arguments = invocation.ArgumentList.Arguments.Skip(1).ToArray();
+			return true;
+		}
+
+		private async Task<bool> IsStringFormatInvocation(Document document, InvocationExpressionSyntax invocation, CancellationToken cancellationToken)
+		{
+			SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken);
```

Comment:
Because getting the semantic model is computationally expensive, we should check for "Format" before, rather than after.